includes/abstracts/llms-abstract-controller-user-engagements.php

```diff
@@ -0,0 +1,253 @@
+<?php
+/**
+ * LLMS_Abstract_Controller_User_Engagements class
+ *
+ * @package LifterLMS/Abstracts/Classes
+ *
+ * @since [version]
+ * @version [version]
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Base class that handles awarded user engagements (achievements and certificates).
+ *
+ * @since [version]
+ */
+abstract class LLMS_Abstract_Controller_User_Engagements {
+
+	use LLMS_Trait_User_Engagement_Type;
+
+	/**
+	 * @since [version]
+	 *
+	 * @var int
+	 */
+	protected const TEXT_SYNC_AWARDED_ENGAGEMENT_INSUFFICIENT_PERMISSIONS = 0;
+
+	/**
+	 * @since [version]
+	 *
+	 * @var int
+	 */
+	protected const TEXT_SYNC_AWARDED_ENGAGEMENT_INVALID_TEMPLATE = 1;
+
+	/**
+	 * @since [version]
+	 *
+	 * @var int
+	 */
+	protected const TEXT_SYNC_AWARDED_ENGAGEMENTS_INSUFFICIENT_PERMISSIONS = 2;
+
+	/**
+	 * @since [version]
+	 *
+	 * @var int
+	 */
+	protected const TEXT_SYNC_AWARDED_ENGAGEMENTS_INVALID_NONCE = 3;
+
+	/**
+	 * @since [version]
+	 *
+	 * @var int
+	 */
+	protected const TEXT_SYNC_MISSING_AWARDED_ENGAGEMENT_ID = 4;
+
+	/**
+	 * @since [version]
+	 *
+	 * @var int
+	 */
+	protected const TEXT_SYNC_MISSING_ENGAGEMENT_TEMPLATE_ID = 5;
```

Comment:
There should be an empty line regardless of whether or not the standards indicate as much.

Reply:
I looked and didn't find summary requirement in the LifterLMS or WordPress documentation standards, but the current version in the proposed [PHP FIG PSR](https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md#51-summary) says:
> A Summary MUST contain an abstract of the "Structural Element" defining the purpose.

I will add summaries to the new constants.

From the [LifterLMS documentation standard](https://github.com/gocodebox/lifterlms/blob/trunk/docs/documentation-standards.md#tag-alignment-and-order):
> All changelog tags, `@since`, `@version`, and `@deprecated` should be grouped together with a space before the first `@since` tag and after the last tag in the group.

Reply:
Fine for the docs standard, only the examples there are misleading. Once again my bad :)